chromadb/test/test_clients.py

```diff
@@ -0,0 +1,79 @@
+# type: ignore
+import chromadb
+import chromadb.server.fastapi
+import pytest
+import tempfile
+
+
+@pytest.fixture
+def ephemeral_api():
+    return chromadb.EphemeralClient()
+
+
+@pytest.fixture
+def persistent_api():
+    return chromadb.PersistentClient(
+        path=tempfile.gettempdir() + "/test_server",
+    )
+
+
+@pytest.fixture
+def persistent_api_cache_bust():
+    return chromadb.PersistentClient(
+        path=tempfile.gettempdir() + "/test_server",
+    )
+
+
+@pytest.fixture
+def http_api():
+    return chromadb.HttpClient()
+
+
+# verify the ephemeral api is set to ephemeral mode (can't persist)
+@pytest.mark.parametrize("api_fixture", [ephemeral_api])
+def test_ephemeral_client(api_fixture, request):
+    api = request.getfixturevalue("ephemeral_api")
+    api.reset()
+    collection = api.create_collection("test")
+    collection.add(ids="id1", documents="hello")
+
+    # should raise a NotImplementedError when calling persist
+    with pytest.raises(NotImplementedError):
```

Comment:
we should just remove persist() from the api, can we find another way to do this check or behvaior?

Reply:
i'd be ok with reaching into the client to check its settings

Reply:
That would be the easiest way. Do you know if that is currently possible? Currently I do not believe that the client has access to settings. `def Client` returns `api: API` ... maybe I'm missing something.